src/py21cmfast/drivers/coeval.py

```diff
@@ -833,17 +823,18 @@ def run_coeval(
             st = st2
 
         perturb_files.append((z, os.path.join(direc, pf2.filename)))
-        if flag_options.USE_HALO_FIELD:
+        if inputs.flag_options.USE_HALO_FIELD:
             hbox_files.append((z, os.path.join(direc, hb2.filename)))
-            pth_files.append((z, os.path.join(direc, ph2.filename)))
-        if flag_options.USE_TS_FLUCT:
+            if not inputs.flag_options.FIXED_HALO_GRIDS:
+                pth_files.append((z, os.path.join(direc, ph2.filename)))
```

Comment:
nitpicky, but I don't love `pth_files` as a name -- I associate `pth` with "path" not perturbed halos.